Add pipeline for British synonyms

beckyjackson opened the pull request:
See #787

Thank you to @matentzn for allowing us to adapt the HPO pipeline for our use 😃 

Please note that the script requires `python3` and the `pandas` module. To install:
```
python3 -m pip install pandas
```

Then, to add synonyms:
```
make add_british_synonyms
```

I've already run it for all terms currently in `doid-edit`, so we can probably just run it every so often to keep things up-to-date. This run added 463 new synonyms.

matentzn commented:
Of course; I do recommend though that the results are eyeballed (ideally the table); we have had some examples (groin, groyne, and others) were blind, decontextualised translation can lead to false synonyms; in this case, it would be good if you could make an HPO ticket to remove the term that resulted in the weird result. We already caught some of those cases - but I am sure there will be others. Similarly, some of the more specialised medical terminology may not yet be reflected - I would also then just append them to the HPO dict if you find them.

lschriml commented:
Hello Becky,
   I like Nico's suggestion, good to be careful.
Can you create a term - synonym table for QC before we add the British
synonyms ?
I can review.

I would like to add just a few of the new British synonyms for this weeks
build, we will do the build Tuesday late afternoon, in preparation for a
website update on Wednesday. Just to check that there are no build or
loading issues with the addition of the synonyms.
   A couple of questions:
      -- As part of this process, have you identified DO terms, where we
already have a
      British version synonym ?
      -- Or DO terms that are the British spelling ?
      -- as part of this process, can you add the US english tag to all
other synonyms ?

   From the list of clinical terms I identified for this update, where the
British spelling is different than the US spelling, will this update add
all of them in one go ?

Cheers,
Lynn

On Sun, Feb 16, 2020 at 10:20 AM Becky Jackson <notifications@github.com>
wrote:

> See #787
> <https://github.com/DiseaseOntology/HumanDiseaseOntology/issues/787>
>
> Thank you to @matentzn <https://github.com/matentzn> for allowing us to
> adapt the HPO pipeline for our use 😃
>
> Please note that the script requires python3 and the pandas module. To
> install:
>
> python3 -m pip install pandas
>
>
> Then, to add synonyms:
>
> make add_british_synonyms
>
>
> I've already run it for all terms currently in doid-edit, so we can
> probably just run it every so often to keep things up-to-date. This run
> added 463 new synonyms.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813
> Commit Summary
>
>    - Add pipeline for British synonyms
>
> File Changes
>
>    - *M* Makefile
>    <https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813/files#diff-0>
>    (37)
>    - *M* src/ontology/doid-edit.owl
>    <https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813/files#diff-1>
>    (463)
>    - *A* src/sparql/doid_labels.rq
>    <https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813/files#diff-2>
>    (11)
>    - *A* src/sparql/doid_synonyms.rq
>    <https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813/files#diff-3>
>    (12)
>    - *A* src/util/compute_british_synonyms.py
>    <https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813/files#diff-4>
>    (54)
>
> Patch Links:
>
>    -
>    https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813.patch
>    - https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DiseaseOntology/HumanDiseaseOntology/pull/813?email_source=notifications&email_token=ABBB4DLIFUIIJSAHMEUNOR3RDFKTJA5CNFSM4KWEDS62YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4IN3I43A>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBB4DJQ7AKLBZHRFPLVSV3RDFKTJANCNFSM4KWEDS6Q>
> .
>


-- 
Lynn M. Schriml, Ph.D.
Associate Professor

Institute for Genome Sciences
University of Maryland School of Medicine
Department of Epidemiology and Public Health
670 W. Baltimore St., HSFIII, Room 3061
Baltimore, MD 21201
P: 410-706-6776 |  F: 410-706-6756
lschriml@som.umaryland.edu

beckyjackson commented:
> Can you create a term - synonym table for QC before we add the British
synonyms ?

I've emailed you the template generated from `make add_british_synonyms`

> I would like to add just a few of the new British synonyms for this weeks
build

Sure. Adding synonyms would not cause an issue with the build, but I understand. I've reverted some changes and only added the first 10 rows from the template.

> As part of this process, have you identified DO terms, where we already have a British version synonym?
> Or DO terms that are the British spelling ?

There is no file output, but this will not duplicate any existing synonyms or labels. So, if a British synonym (or label) already exists with the same spelling as the American-British table, then nothing will be added. That said, I ran a "backwards" version of the script and there were no results, so it looks like we don't have any British spellings in labels/synonyms currently.

> as part of this process, can you add the US english tag to all other synonyms ?

Yep, I just added this to the PR.
